firefox_puppeteer/tests/test_tabs.py

```diff
@@ -2,78 +2,152 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from marionette import Wait
-from marionette.errors import NoSuchElementException
-
 from firefox_ui_harness.testcase import FirefoxTestCase
 
 
-class TestTabs(FirefoxTestCase):
+class TestTabBar(FirefoxTestCase):
+
+    def test_basics(self):
+        tabbar = self.browser.tabbar
+
+        self.assertEqual(tabbar.window, self.browser)
+
+        self.assertEqual(len(tabbar.tabs), 1)
+        self.assertEqual(tabbar.tabs[0].handle, self.marionette.current_window_handle)
+
+        self.assertEqual('toolbarbutton', tabbar.newtab_button.get_attribute('localName'))
+        self.assertEqual('tabs', tabbar.toolbar.get_attribute('localName'))
+
+    def test_open_close(self):
+        tabbar = self.browser.tabbar
+
+        self.assertEqual(len(tabbar.tabs), 1)
+        self.assertEqual(tabbar.selected_index, 0)
+
+        # Open with default trigger, and force closing the tab
+        tabbar.open_tab()
+        tabbar.close_tab(force=True)
+
+        # Open a new tab by each trigger method
+        open_strategies = ('button',
+                           'menu',
+                           'shortcut',
+                           lambda tab: tabbar.newtab_button.click()
+                           )
+        for trigger in open_strategies:
+            new_tab = tabbar.open_tab(trigger=trigger)
+            self.assertEqual(len(tabbar.tabs), 2)
+            self.assertEqual(new_tab.handle, self.marionette.current_window_handle)
+
+            tabbar.close_tab()
+            self.assertEqual(len(tabbar.tabs), 1)
+            self.assertNotEqual(new_tab.handle, self.marionette.current_window_handle)
```

Comment:
Depending on settings in Firefox there is a different behavior in closing tabs. It could be right to left or the other way around. We could combine the proposal from above with this strategy then.